GameObject -> BloonScript reference

SSeelos opened the pull request:
you can reference monobehaviours instead of gameobjects so you dont have to call get component (wich might not even be on it)

Commit:
GameObject -> BloonScript reference

you can reference monobehaviours instead of gameobjects so you dont have to call get component (wich might not even be on it)

Serpendio commented:
What if for this one I know that I wish to also instantiate with the Rigidbody & SpriteRenderer? 
Sure, I could AddComponent, but using a prefab is both simpler to quickly modify and less memory intensive. No?

SSeelos commented:
that should not be a problem. the reference will still point to the prefab where the monobehaviour is added to. so you can just call get component on the component itself just as you would with a gameobject reference
![image](https://github.com/Serpendio/TotallyOriginalTowerDefense/assets/30175883/b741a2ee-caff-4ff1-9aa7-00364fcdb35f)

Commit:
scripttemplates

keep in mind that if you change or add scripttemplates you have to restart the editor for the changes to take effect

Serpendio committed:
Update 81-C# Script__MonoBehaviour-MB.cs.txt

Not all MonoBehaviours even need awake or system.collections, so removed those from the template file

Commit:
Merge pull request #3 from Serpendio/Serpendio-patch-1

Update 81-C# Script__MonoBehaviour-MB.cs.txt

SSeelos commented on Assets/Scripts/BloonSpawner.cs:
```diff
@@ -48,8 +48,8 @@ public void SpawnWave()
-        Instantiate(bloonBase);
-        var script = bloonBase.GetComponent<BloonScript>();
+
+        var script = Instantiate(bloonBase);
```
my review

SSeelos replied:
my reply

SSeelos reviewed (commented):
my comment

SSeelos commented on Assets/ScriptTemplates/81-C# Script__MonoBehaviour-MB.cs.txt:
```diff
@@ -1,13 +1,8 @@
     #ROOTNAMESPACEBEGIN#
 public class #SCRIPTNAME# : MonoBehaviour
 {
-    void Awake()
```
i like having awake in there but i see your point
i use awake to set up dependencies (getcomponent) calls
then in start i might get properties of the cached components

Serpendio commented on Assets/ScriptTemplates/81-C# Script__ScriptableObject-SO.cs.txt:
```diff
@@ -0,0 +1,11 @@
+using System.Collections;
+using System.Collections.Generic;
```
```suggestion
```
No need for these in most ScriptableObjects either

Commit:
Update Assets/ScriptTemplates/81-C# Script__ScriptableObject-SO.cs.txt

jep

Co-authored-by: Serpendio <58090288+Serpendio@users.noreply.github.com>
